mdtraj/core/topology.py

```diff
@@ -84,6 +85,13 @@ def _topology_from_subset(topology, atom_indices):
     atom_indices : array_like, dtype=int
         The indices of the atoms to keep
     """
+    np_indices = np.array(atom_indices)
+    if len(np_indices) > 1:
+        if (np_indices[1:] - np_indices[:-1]).min() < 1:
+            warnings.warn('atom_indices are not monotonically increasing')
+        if len(np.unique(np_indices)) < len(np_indices):
+            warnings.warn('atom_indices are not unique')
```

Comment:
Could you provide some rough benchmarks that look at how much this slows down common operations (i.e. loading topologies) for a few different-sized systems? These checks should be quick but they're O(N) and called in most user workflows

Reply:
Using the topologies from the AMBER20 Benchmark suite (http://ambermd.org/GPUPerformance.php)  I get the following figures for mdt.load_topology():

System         n_atoms    1.9.7       #1753
STMV            23558        0.248       0.285
Factor IX      90906        0.807       0.880
JAC              408609        3.804       3.712
Cellulose 1067055         9.590      9.567

Times (in seconds) are averages over 5 repeats, jobs run on my intel i5 laptop

Reply:
Interesting ... I can't imagine this change would actually speed anything up so it's a bit of a surprise that some runs a faster. I suppose the slowdown is simply less (much less) than the variance of 5 runs.

Reply:
Exactly - I didn’t actually calculate the variance but as I recall there was a fair bit of jitter between runs - these are wallclock times by the way, so that could account for some of it.